### PR TITLE
fix(core): clone system properties before iterating

### LINKED ISSUE #1023: ConcurrentModificationException while iterating through system properties
**Expected behavior**

Create an Atomix builder.

**Actual behavior**
On `Atomix.builder()`, a ConcurrentModificationException is thrown when iterating through system properties. This happens randomly during tests that have multiple instances of Atomix running. Stacktrace:
```
java.util.ConcurrentModificationException
	at java.util.Hashtable$Enumerator.next(Hashtable.java:1387)
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.noneMatch(ReferencePipeline.java:459)
	at io.atomix.core.utils.config.PolymorphicConfigMapper.lambda$checkRemainingProperties$7(PolymorphicConfigMapper.java:100)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1556)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.atomix.core.utils.config.PolymorphicConfigMapper.checkRemainingProperties(PolymorphicConfigMapper.java:102)
	at io.atomix.utils.config.ConfigMapper.map(ConfigMapper.java:173)
	at io.atomix.utils.config.ConfigMapper.map(ConfigMapper.java:134)
	at io.atomix.utils.config.ConfigMapper.loadResources(ConfigMapper.java:124)
	at io.atomix.utils.config.ConfigMapper.loadFiles(ConfigMapper.java:78)
	at io.atomix.core.Atomix.config(Atomix.java:320)
	at io.atomix.core.Atomix.builder(Atomix.java:346)
	at io.atomix.core.Atomix.builder(Atomix.java:332)
```

**Steps to reproduce**

I wasn't able to reproduce the issue in a deterministic way. 

**Minimal yet complete reproducer code (or URL to code)**

```java
    final AtomixBuilder atomixBuilder =
        Atomix.builder()
            .withClusterId("dl-test")
            .withMemberId(String.valueOf(nodeId))
            .withAddress(Address.from(socketAddress.host(), socketAddress.port()));
```

**Environment**

 - Atomix: master
 - OS: `4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
 - `openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-2ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)`

<sup>:warning:️ Please verify that your issue still occurs on the latest version of Atomix before reporting. 
The documentation is currently work-in-progress and is not yet complete.</sup>

-----------------------------------------------------------------------------------------
*Have you searched the CLOSED issues already? How about checking in with the [Atomix Google Group](https://groups.google.com/forum/#!forum/atomixio)?*

### PR DESCRIPTION
**Pull request by MiguelPires**
Closes #1023

### CODE REVIEW

File: core/src/main/java/io/atomix/core/utils/config/PolymorphicConfigMapper.java [hunk 1]
```diff
@@ -97,7 +98,7 @@ protected void checkRemainingProperties(Set<String> missingProperties, List<Stri
         .filter(propertyName -> !isPolymorphicType(clazz) || !polymorphicTypes.stream().anyMatch(type -> Objects.equals(type.getTypePath(), propertyName)))
         .map(propertyName -> toPath(path, propertyName))
         .filter(propertyName -> !properties.containsKey(propertyName))
```
Comment (johnou) [inline core/src/main/java/io/atomix/core/utils/config/PolymorphicConfigMapper.java]: properties is also accessed in the filter call above, wouldn't it make sense making a new Properties above? eg..

```java
Properties localProperties = new Properties();
localProperties.putAll(System.getProperties());
```

Comment (MiguelPires) [reply core/src/main/java/io/atomix/core/utils/config/PolymorphicConfigMapper.java]: You're right, thanks

**Comment by johnou** [comment]
seems docker build is failing..

```
[INFO] W: Failed to fetch http://deb.debian.org/debian/dists/jessie-backports/main/binary-amd64/Packages  404  Not Found
[INFO] 
[INFO] E: Some index files failed to download. They have been ignored, or old ones used instead.
[INFO] [0m
[ERROR] The command '/bin/sh -c apt-get update -y && apt-get install -y iptables stress' returned a non-zero code: 100
```

cc @kuujo

**Comment by MiguelPires** [comment]
Thanks for the review @johnou. I squashed my commits, let me know if there's anything else I need to do for the PR to be merged

**Comment by johnou** [comment]
@MiguelPires please rebase so CI passes.

**Comment by johnou** [comment]
Thanks!

**Pull request merged**